clients/src/main/java/org/apache/kafka/common/errors/OffsetOutOfRangeException.java

```diff
@@ -35,4 +43,8 @@ public OffsetOutOfRangeException(String message, Throwable cause) {
         super(message, cause);
     }
 
+    public Map<TopicPartition, Long> getOutOfRangePartitions() {
```

Comment:
Thanks @lindong28.

Regarding the final field in `OffsetOutOfRangeException`, it's a good question as there is no nice solution. :) Ideally, we would not have 4 constructors since only 1 or 2 tend to be used, but too late to change that now. I think it's OK to leave it as you have as I don't know how committers feel about this particular issue (and I don't want to steer you towards a direction that they may disagre with!).